sycl/include/CL/sycl/detail/scheduler/scheduler.cpp

```diff
@@ -63,7 +63,7 @@ void Node::addAccRequirement(
     int argIndex) {
   detail::buffer_impl<dataT, dimensions> *buf =
       Acc.template accessor_base<dataT, dimensions, accessMode, accessTarget,
-                                 isPlaceholder>::__impl()
+                                 isPlaceholder>.__impl()
```

Comment:
I think the right fix is to remove the declaration of the base class. I.e.
```c++
detail::buffer_impl<dataT, dimensions> *buf = Acc.__impl()->m_Buf;
```
Otherwise I see compilation errors from the GCC 5.4 about wrong usage of accessor_base class.
Could you check if proposed change fixes the build with GCC 8, please?